eng/testing/tests.mobile.targets

```diff
@@ -232,9 +232,26 @@
   <Import Project="$(MonoProjectRoot)\wasm\build\WasmApp.InTree.targets" Condition="'$(TargetOS)' == 'Browser'" />
   <PropertyGroup>
       <WasmBuildAppDependsOn>PrepareForWasmBuildApp;$(WasmBuildAppDependsOn)</WasmBuildAppDependsOn>
+      <EmSdkDir>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'src', 'mono', 'wasm', 'emsdk'))</EmSdkDir>
   </PropertyGroup>
 
-  <Target Condition="'$(TargetOS)' == 'Browser'" Name="BundleTestWasmApp" DependsOnTargets="WasmBuildApp" />
+  <Target Condition="'$(TargetOS)' == 'Browser'" Name="BundleTestWasmApp" DependsOnTargets="WasmBuildApp;StageEmSdkForHelix" />
+
+  <!-- on the CI, we don't have emscripten provisioned in src/mono/wasm/emsdk. Instead we have
+    the path in $(EMSDK_PATH). But we can't use that for packaging emsdk as a helix correlation
+    payload because the helix tasks try to wrote a .payload file to that directory, which
+    we can't because it could be in `/usr/local/emscripten`. So, we copy the files over to
+  `src/mono/wasm/emsdk`, and use that. -->
```

Comment:
Does the value of `$(EMSDK_PATH)` in CI vary? The wording/train of thought was a bit confusing to me, does this make sense?

```suggestion
  <!-- CI has emscripten provisioned in $(EMSDK_PATH) as `/usr/local/emscripten`. Because helix tasks will attempt to write a .payload file, we cannot use $(EMSDK_PATH) to package emsdk as a helix correlation payload. Instead, we copy over the files to a new directory `src/mono/wasm/emsdk` and use that. -->
```

Reply:
Yeah, AFAICS, on the CI we don't use `provision-wasm`, so it's not present in `$(RepoRoot)/src/mono/wasm`. Instead, it is available in `/usr/local/emscripten`.